src/app/components/content-creator-page/content-creator-page.component.spec.ts

```diff
@@ -121,104 +120,186 @@ describe('ContentCreatorPageComponent', () => {
   });
   // =====End Element creation finds=====
 
-  /**
```

Comment:
How come all relevant comments have been removed? If there have been changes which make the comment incorrect, it should be updated. There is no need to remove entirely. This whole file has a lot more tests, which are good, but all documentation is now gone. This needs documentation.

Reply:
Hello @IkenoXamos ,  I removed the comments to adhere to best practice. As stated by various developers and by Carolyn, unit test names should be explanatory in themselves. comments should be added mainly where additional information may be necessary. The first argument in a unit test function, the string, should clearly describe what a test is meant to do. The additional comments may have been necessary when previous tests had multiple assertions but having multiple assertions in unit tests is itself also bad practice. Looking at the names I have added, I realize the string argument can be written better. However, by adhering to best practice, I still do not see the need for a lot of the comments. In addition to this, if the comments were to be kept, I learned that they should be used as document comments rather than pseudo like comments. Like below:
/**
  * Gets script version
  * @param fileName
  * @returns script version
  */
A good example of why I removed comments is the comment "// Test radio selformat" which is placed on top of an assertion in a method that has multiple assertions. FIrst I split these assertions into multiple methods then I added the description in it. So there is no point in having a comment on that. 
Most of my references are documentation and/or Carolyn. All unit tests should adhere to the FIRST principle. One of the principles states "A test method should test for a single logical outcome, implying that typically there
should be only a single logical assert.". Following that logic, and usage of the first parameter in "it", there seems to be little need for a majority of the comments that I saw. 
In the comments, the use of the words "now", "first", "next", implies an ordering to the tests. However tests should run independent of ordering. That may not be what was intended but simply how I perceived it.
As I previously stated, I do believe my test descriptions could possibly be written better, so I can look into renaming them.

Reply:
Yes, that in general is all fine. I agree with those best practices for the most part. However, I suppose I am more of an opinion that some header comments using the /** are acceptable. I wasn't suggesting that each individual it() has its own header comment either. But, some of your tests were not quite as simple as a single assert, and used external libraries such as spyOn. These libraries are common, so not a whole lot of explanation would be needed, but a little bit is helpful. There are some test files with hundreds of lines of tests, and no extra documentation outside of the it() statements. Even if those it() statements were clarified, there are definitely a good several spots where even further documentation in the form of /** comments would be very helpful.

Reply:
This is a good point with the external libraries. There is a testing section in the wiki, maybe we can include some general documentation there to help shed some light on how additional libraries were used for future associates.

Reply:
The wiki doc sounds good. Documenting when it is necessary to use fake async and tick to handle the microtasks would probably be good since there is a constant shift between promises and observables within the code. Also when Michael added the Mat* related libraries, I ended up needing to pass in custom spy objects into the provider's values in order to properly use the dependencies for testing. If there is a location that requires specific documentation, I will gladly add it.